docs/adr/0005-rapid-releases-and-moving-towards-a-monorepo.md

```diff
@@ -0,0 +1,83 @@
+  - Releases require too much developer involvement.  Normal releases take a couple hours, during which a developer needs to monitor the progress and perform several manual steps along the way.  If the unreleased code has breaking changes, then the developer also needs to merge the PR for the consumer application(s) after the release, which is not ideal since that developer is often not the one who wrote the PR.  Worse, if the dev who caused the breaking changes forgot to create a PR, they need to create that PR now, leading to multiple devs coordinating for the release.  Releases that are not from main also require extra work: creating a branch for that version, backporting the fix, making a patch release, etc.
+  - The mobile teams are moving towards a weekly release cycle, which may lead to us needing to do more releases for older versions in order to uplift fixes.
+
+At the same time, Mozilla is in the process of moving towards a single monorepo.  In a monorepo, the application-services code will be immediately consumed by all applications as soon as it's merged.  This is drastically different than our current workflow, where code sits in main for long periods of time before being released, then sits in a release for a long period of time before being taken by all consumers.
+
+We have an opportunity to update our release process to follow the rapid release model and to be more compatible with the future monorepo world.
+
+## Decision Drivers
+
+* We want to get changes merged and released to our consumer applications faster
+* We want to move towards the future monorepo world
+* We are willing to accept the trade-offs involved with more frequent merges to application repositories.
+
+## Considered Options
+
+* **(A) Keep our current release process**
+    * Keep everything the same for a while and wait for other code to move into the monorepo.
+* **(B) Move to a rapid release process where changes in app-services are frequently merged into consumer applications**
+    * Update our branches to match how moz-central: we would have a `main`, `beta` and `release` branch, with nightly being built off of main, as well as `esr-*` branches for older ESR releases.
+    * Change it so our code lives in the repositories for consumer applications including firefox-android and firefox-ios.  We could either use a git submodule or vendor a copy of our code in to accomplish this.  Note: This is already planned for firefox-ios.
+    * Automate a nightly merge of our code our `main` branch to the `main` branch of the consumer app repos:
+       * Automate an application-services commit that updates our version number to something like `[firefox-version].nightly-[YYYYMMDD]`
```

Comment:
It's the application-services version that the applications consume -- i.e. it would replace the `96.1.3` version that we're currently on.   We could also make a github tag, that makes sense to me.  And this would definitely be before the PRs.